apps/web/src/components/Files.tsx

```diff
@@ -424,8 +424,12 @@ function FileItem(props: FileItemProps) {
     props.onUseInSQL(props.file)
   }, [props.onUseInSQL, props.file])
 
-  const onRemove = useCallback(() => {
-    props.onDelete(props.file)
+  const [isDeleting, setIsDeleting] = useState(false)
+
+  const onRemove = useCallback(async () => {
+    setIsDeleting(true)
+    await props.onDelete(props.file)
+    setIsDeleting(false)
```

Comment:
I think we should use the existing `isDeleting` prop.
What we need to do here is debug why it is not becoming `true` when the data source is being removed.

We should check the parent of this component, check what it is giving as the `isDeleting` prop and why it is not passing `true` while the file is being deleted.